data_chunk.go

```diff
@@ -39,23 +41,48 @@ func (chunk *DataChunk) SetSize(size int) error {
 
 // GetValue returns a single value of a column.
 func (chunk *DataChunk) GetValue(colIdx, rowIdx int) (any, error) {
-	if colIdx >= len(chunk.columns) {
+	if chunk.projection == nil && (colIdx < 0 || colIdx >= len(chunk.columns)) {
 		return nil, getError(errAPI, columnCountError(colIdx, len(chunk.columns)))
 	}
-	column := &chunk.columns[colIdx]
 
+	if chunk.projection != nil && (colIdx < 0 || colIdx >= len(chunk.projection)) {
+		return nil, getError(errAPI, columnCountError(colIdx, len(chunk.projection)))
+	}
+
+	// Rewrite colIdx for projection, if provided. Otherwise, use as is for bwc.
```

Comment:
nit: can we write out `bwc` - it took me a second to understand what it took for 😅